Sources/SemanticAnalyzer/SemanticAnalyzer+Statements.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  SemanticAnalyzer+Statements.swift
+//  SemanticAnalyzer
+//
+//  Created by Farcas, Calin on 10/11/2018.
+//
+import Foundation
+import AST
+import Lexer
+import Diagnostic
+
+extension SemanticAnalyzer {
+  public func postProcess(ifStatement: IfStatement,
+                          passContext: ASTPassContext) -> ASTPassResult<IfStatement> {
+    let ifStatement = ifStatement
+    let condition = ifStatement.condition
+    let environment = passContext.environment!
+    let enclosingTypeIdentifier = passContext.enclosingTypeIdentifier!
+    let typeStates = passContext.contractBehaviorDeclarationContext?.typeStates ?? []
+    let callerProtections = passContext.contractBehaviorDeclarationContext?.callerProtections ?? []
+    var diagnostics = [Diagnostic]()
+
+    // Allow let statements
+    switch condition {
+    case .variableDeclaration(let variableDeclaration):
```

Comment:
Why is this separate from the `.binaryExpression` case? What exactly does it handle? We shouldn't allow e.g. `if let x: Int { ... }` since it makes no sense.

Reply:
I agree with this, although it wasn't very clear from the ticket (I'll change it). As a side note, it doesn't make very much sense to me that let statements are parsed as binary expressions if they're not statements, but I went with this because it might break a lot of things if I change it (worth investigating in the future).